renormalizer/mps/mps.py

```diff
@@ -1240,6 +1240,156 @@ def full_wfn(self) -> np.array:
             dim2 = mt.shape[-1]
             res = np.tensordot(res, mt.array, axes=1).reshape(1, dim1, dim2)
         return res[0, :, 0]
+    
+    def calc_1ordm(self):
+        r""" Calcuate 1-orbital reduced density matrix
+        
+            .. math::
+                \rho_i = \textrm{Tr}_{j \neq i} | \Psi \rangle \langle \Psi |
+        
+        Returns
+        -------
+        rdm: Dict
+            {0:\rho_0, 1:\rho_1, ...}. The key is the index of the site.
+        """
+
+        identity = Mpo.identity(self.mol_list)
+        environ = Environ(self, identity, "R")
+        
+        rdm = {}
+        for ims, ms in enumerate(self):
+            ltensor = environ.GetLR(
+                "L", ims-1, self, identity, itensor=None, method="System"
+            )
+            rtensor = environ.GetLR(
+                "R", ims+1, self, identity, itensor=None, method="Enviro"
+            )
+            ltensor = ltensor.reshape(ltensor.shape[0], ltensor.shape[-1])
+            rtensor = rtensor.reshape(rtensor.shape[0], rtensor.shape[-1])
+            
+            tensor = tensordot(ltensor, ms.conj(), ([0],[0]))
+            tensor = tensordot(tensor, rtensor, ([-1],[0]))
+            if ms.ndim == 3:
+                tensor = tensordot(tensor, ms, ([0,-1],[0,-1]))
+            else:
+                tensor = tensordot(tensor, ms, ([0,-1,-2],[0,-1,-2]))
+            assert xp.allclose(tensor, tensor.T.conj())
+            rdm[ims] = asnumpy(tensor)
+
+        return rdm
+    
+    def calc_2ordm(self):
+        r""" Calcuate 2-orbital reduced density matrix
+        :math: `\rho_{ij} = \textrm{Tr}_{k \neq i, k \neq j} | \Psi \rangle \langle \Psi |`
```

Comment:
This equation fails to render in the doc.

Reply:
I will check it.

Reply:
The problem is probably because of the additional space in `:math: `\rho_{ij}`